src/main/java/xyz/nucleoid/plasmid/game/GameType.java

```diff
@@ -21,19 +21,10 @@
  * @param <C> the type of config that should be loaded
  * @see GameConfig
  */
-public final class GameType<C> {
+public record GameType<C>(Identifier id, Codec<C> configCodec,
+                          xyz.nucleoid.plasmid.game.GameType.Open<C> open) {
     public static final TinyRegistry<GameType<?>> REGISTRY = TinyRegistry.create();
 
-    private final Identifier id;
-    private final Codec<C> configCodec;
-    private final Open<C> open;
-
-    private GameType(Identifier id, Codec<C> configCodec, Open<C> open) {
```

Comment:
Likewise here, the GameType must not be constructable without calling the intended register function